dnf-behave-tests/dnf/api/transaction-test.feature

```diff
@@ -0,0 +1,30 @@
+@dnf5
+Feature: transaction: dry-run a transaction
+
+
+@bz2109660
+Scenario: Test labirinto install transaction when it should succeed
+Given I use repository "simple-base"
+ When I execute python libdnf5 api script with setup
+      """
+      goal = libdnf5.base.Goal(base)
+      goal.add_rpm_install("labirinto")
+      assert test_transaction(goal) == libdnf5.base.Transaction.TransactionRunResult_SUCCESS
+      """
+ Then the exit code is 0
+  And stderr does not contain "Traceback"
+
+@bz2109660
+Scenario: Test labirinto install transaction when it should fail
+Given I use repository "simple-base"
+ When I execute python libdnf5 api script with setup
+      """
+      goal = libdnf5.base.Goal(base)
+      goal.add_rpm_install("labirinto")
+      execute_transaction(goal, "install a package")
+
+      # We have already run the transaction, so test_transaction should return a failing status
+      assert test_transaction(goal) == libdnf5.base.Transaction.TransactionRunResult_ERROR_RPM_RUN
+      """
+ Then the exit code is 0
+  And stderr does not contain "Traceback"
```

Comment:
We would like to have the tests as strict as possible therefore I think it would be better to use
```gherkin
And stderr is empty
```
The stderr really should be empty, no problems are expected here right?
(for both tests)

Reply:
I've just run the test on f38 with dnf5-5.0.6-1.fc39, the first scenario passed as it is and also with "stderr is empty". The second scenario failed due to `AttributeError: 'Transaction' object has no attribute 'set_description'`, the build probably doesn't contain related PR.

Reply:
> The stderr really should be empty, no problems are expected here right?
Yes, thanks, done!

Reply:
> I've just run the test on f38 with dnf5-5.0.6-1.fc39, the first scenario passed as it is and also with "stderr is empty". The second scenario failed due to `AttributeError: 'Transaction' object has no attribute 'set_description'`, the build probably doesn't contain related PR. When dnf5-5.0.6-0.20230220010121.5.0.5+54.g9a7b19b1.fc39 is used instead, both scenarios work OK, also with "stderr is empty".

Yes, see https://github.com/rpm-software-management/ci-dnf-stack/pull/1221